### PR TITLE
cmd/kubectl-gadget: Uniform event-based gadgets

### PR DESCRIPTION
**Pull request by blanquicet**
# Uniform how event-based gadgets print the events

This PR fixes https://github.com/kinvolk/inspektor-gadget/issues/546 by uniforming all the event-based gadgets: all the gadgets in the `trace` category, the `audit-seccomp` and the gadgets. I also aligned the `snapshot` category following the same approach.

It also fixes some minor issues like:

- Add a newline when printing the special events (ERR, WARN, etc).
- Do not include the mount ns id on the process gadget to make the output of the JSON and the columns aligned.

## Note
This is the first step to create common packages that do not depend on the Trace CRD (https://github.com/kinvolk/inspektor-gadget/issues/642).

### CODE REVIEW

File: cmd/kubectl-gadget/snapshot/socket.go [hunk 1]
```diff
@@ -39,96 +40,18 @@ var socketCollectorCmd = &cobra.Command{
 	Short: "Gather information about TCP and UDP sockets",
 	RunE: func(cmd *cobra.Command, args []string) error {
 		callback := func(results []gadgetv1alpha1.Trace) error {
-			allSockets := []socketcollectortypes.Event{}
+			allSockets := []types.Event{}
 
 			for _, i := range results {
-				var sockets []socketcollectortypes.Event
+				var sockets []types.Event
 				json.Unmarshal([]byte(i.Status.Output), &sockets)
```
[the remark this answers is not in the document]
Comment (blanquicet) [reply cmd/kubectl-gadget/snapshot/socket.go:47]: Added

File: pkg/gadgets/process-collector/tracer/tracer.go [hunk 2]
```diff
@@ -113,10 +113,9 @@ func RunCollector(resolver gadgets.Resolver, node, mntnsmap string) ([]processco
 				Pod:       container.Podname,
 				Container: container.Name,
 			},
-			Tgid:      tgid,
-			Pid:       pid,
-			Command:   command,
-			MountNsID: mntnsid,
+			Tgid:    tgid,
+			Pid:     pid,
+			Command: command,
```
[the remark this answers is not in the document]
Comment (blanquicet) [reply pkg/gadgets/process-collector/tracer/tracer.go:118]: Oh, I didn't realize it was also present on other gadgets. Yeah, I will keep it and update the doc. My only concern is that the information showed in columns will be different from the one with JSON mode.

File: cmd/kubectl-gadget/utils/events.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	if e.Namespace != "" && e.Pod != "" {
+		podMsgSuffix = ", pod " + e.Namespace + "/" + e.Pod
+	}
+
+	if e.Type == eventtypes.DEBUG && !verbose {
+		return
+	}
+
+	fmt.Fprintf(os.Stderr, "%s: node %s%s: %s\n", e.Type, e.Node, podMsgSuffix, e.Message)
+	return
```
[the remark this answers is not in the document]
Comment (blanquicet) [reply cmd/kubectl-gadget/utils/events.go:41]: I changed it a couple of times, but the result didn't convince me either. I will move to the switch. Thanks!

**Comment by blanquicet** [comment]
> It looks great, very nice cleanup!
>
Yeah, I also like the result. With this PR, we will have the source code of all the gadgets within the `trace` and `snapshot` categories uniform among them. And, in addition, all the "event-based" categories (`trace`, `audit` and `snapshot`) follow now the same approach to printing the event. 

The next step is to remove all the duplicated code and create utils functions for the common patterns, as I did for the management of the special events. And then, finally, work on https://github.com/kinvolk/inspektor-gadget/issues/642 🙂 .

**Comment by blanquicet** [comment]
I managed all the comments I added the last commit to uniform the gadgets under the `top` category.

**Review by mauriciovasquezbernal — APPROVED** [review]
LGTM. Thanks for working on this.

**Pull request merged**